packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx

```diff
@@ -105,38 +328,393 @@ export class SankeyChartBase extends React.Component<
 
   private _createNodes(width: number): React.ReactNode[] | undefined {
     const nodes: React.ReactNode[] = [];
+
     if (this.props.data.SankeyChartData) {
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
       this.props.data.SankeyChartData.nodes.forEach((singleNode: any, index: number) => {
+        const onMouseOut = () => {
+          this._onLeave(singleNode);
+        };
         const height = singleNode.y1 - singleNode.y0 > 0 ? singleNode.y1 - singleNode.y0 : 0;
+        let padding = 8;
+        const calloutProps = {
+          isCalloutVisible:
+            this.state.selectedState &&
+            this.state.selectedNode &&
+            this.state.selectedNode!.y1 - this.state.selectedNode!.y0 < MIN_HEIGHT_FOR_TYPE,
+          directionalHint: DirectionalHint.topAutoEdge,
+          id: `toolTip${getId('nodeGElement')}`,
+          target: this.state.nodeElement,
+          isBeakVisible: false,
+          gapSpace: 15,
+          onDismiss: this._onLeave,
+          preventDismissOnLostFocus: true,
+        };
+        let textLengthForNodeWeight = 0;
+
+        // If the nodeWeight is in the same line as node description an extra padding
+        // of 6 px is required between node description and node weight.
+        if (height < MIN_HEIGHT_FOR_DOUBLINE_TYPE) {
+          padding = padding + 6;
+          const tspan = select('.nodeName').append('text').attr('class', 'tempText').append('tspan').text(null);
+          tspan.text(singleNode.actualValue);
+          if (tspan.node() !== null) {
+            textLengthForNodeWeight = tspan.node()!.getComputedTextLength();
+            padding = padding + textLengthForNodeWeight;
+          }
+          tspan.text(null);
+          selectAll('.tempText').remove();
+        }
+
+        const truncatedname: string = this._truncateText(singleNode.name, 116, padding);
```

Comment:
what is 116

Reply:
Since the total width of the node is 124 and we are giving margin of 8px from the left . So the actual value on which it will be truncated is 124-8=116.

Reply:
This is done.